ui/packages/consul-ui/app/routes/dc/peers.js

```diff
@@ -2,10 +2,10 @@ import { inject as service } from '@ember/service';
 import Route from '@ember/routing/route';
 
 export default class PeersRoute extends Route {
-  @service features;
+  @service abilities;
 
   beforeModel() {
-    if (!this.features.isEnabled('peering')) {
+    if (!this.abilities.can('use peers')) {
```

Comment:
Throwing a 404 error results in the following UI:

<img width="1225" alt="Screenshot 2022-07-12 at 10 32 28" src="https://user-images.githubusercontent.com/242299/178446881-91c38d04-4cf0-45b6-a693-782a3dd235d4.png">


The consul logo is the `BrandLoader` that is still visible because its opacity does not get set to 0 if we are displaying an error on the top level of the application as the top-level outlet does not get rendered which is responsible to hide the brand-loader.

I moved the check for the peers feature into the peers template and am rendering an AppError component now instead - which results in the following UI.

<img width="1223" alt="Screenshot 2022-07-12 at 11 22 30" src="https://user-images.githubusercontent.com/242299/178457276-15430ed1-89ef-49c4-a962-5ee9d66b8c20.png">


This is necessary as the application doesn't follow regular ember-semantics when it comes to error handling with an error route but uses the top-level `error`-property on the application controller to display an AppError instead. This is suboptimal as there's no great way to unset that error anywhere because of the custom outlet handling Consul uses.

Reply:
We spoke offline about this and we decided we were going to just move peering into a "subapp" like nspaces and partitions, which then means that these routes just never exist at all and therefore naturally 404